### ISSUE TITLE
Issue with chapter 15 Sentiment Analysis with LSTM. Getting 100% accuracy while training.

### ISSUE THREAD

**Issue by sabre-code** [rasbt/machine-learning-book#135]
Hello @rasbt. 
I am facing some issue with chapter 15 part 2 notebook. I tried running the notebook on colab.
Issues:
1. According to the book the IMDB test data should have 25k samples. Currently the IMDB test data has 12.5k samples.
2. Need for installation of portalocker.
3. While training I am getting 100% validation accuracy right from from first epoch. (100% accuracy for all further epochs), and the test accuracy is also 100%.
![Screenshot 2023-07-15 175804](https://github.com/rasbt/machine-learning-book/assets/64346270/98cf5863-9da2-4a9e-9ebb-34ff081d08dd)

**Comment by rasbt** [comment]
Hi there,

I think that's because there were a few changes in torchtext. In order to reproduce the results in the book, you can try 

```
pip install torchtext==0.10.0
```

to install the version I used in the book from https://pypi.org/project/torchtext/0.10.0/

**Comment by sabre-code** [comment]
Hello I just tried installing the said version but torch text 0.10.0 seems unavailable.

**Comment by rasbt** [comment]
I can confirm. It jumps from 0.6 to 0.12. They must have removed this version. This is so weird. It's still available online but it seems that there is no source file anymore: https://pypi.org/project/torchtext/0.10.0/#files

 
<img width="1281" alt="Screenshot 2023-07-15 at 11 33 55 AM" src="https://github.com/rasbt/machine-learning-book/assets/5618407/17bb8ac3-3f99-43b0-8e0e-bee7378fd4b4">

**Comment by sabre-code** [comment]
I guess changing the code to suite the new torchtext is only option. Will try that.

**Comment by rasbt** [comment]
Yeah, that might be the unfortunate consequence of that. Sorry about that! I am a bit hesitant to update the original notebook since it would then be different from the book text, which could be confusing for some. But if you get it to work with a newer version, I'd appreciate a PR. We could include this as an alternative notebook here on GitHub and then link it from the original one.

**Comment by sabre-code** [comment]
Surely I will work on it.

**Comment by sabre-code** [comment]
Hello, I have created PR please check.
